### DISCUSSION TITLE
setPermissionsTeamId

### DISCUSSION THREAD

**Discussion by talovicnedim** [spatie/laravel-permission#2146] [Q&A] [answered]
Hi everyone,

We have multiple workspaces under one account and we added workspace_id in the database.

This is how we see if user has an access to workspace 1:

```
$user = User::query()->where('id', 1)->firstOrFail();

$workspaceId = 1;

setPermissionsTeamId($workspaceId);

$user->hasPermissionTo('Full Access');
```

It works well. However, if we want to check permissions for multiple workspaces at once, then we have the issue:

```
$user = User::query()
          ->where('id', $userId)
          ->firstOrFail();

$workspaces = Workspace::query()
    ->select('id')
    ->where('company_id', auth()->user()->company_id)
    ->get();

return collect($workspaces)
    ->mapWithKeys(function ($workspace) use ($userId) {
       return [
           $workspace->id => $this->hasAccessTo($user, $workspace->id),
       ];
    })
    ->toArray();
```

```
protected function hasAccessTo(User $user, int $workspaceId): bool
{
    setPermissionsTeamId($workspaceId);

    return $user->hasPermissionTo('Full Access');
}
```

This doesn't work. Unless we put `$user` query in the `hasAccessTo` method.

I'm wondering what might be the best option to make it works. 

Adding a query in the `hasAccessTo` is something I'd like to avoid although there will not be so many requests so it's fine to put it there too.

Another option would be to make a custom query but I'd like somehow make it works with **package methods**.

Thanks!

EDIT:

Found the answer here:
https://github.com/spatie/laravel-permission/discussions/2136

**Comment by erikn69** [comment, marked as answer]
Yes, always read other issues or discussions first before opening a new one
https://github.com/spatie/laravel-permission/discussions/2136

**Reply by talovicnedim** [reply]
Yes, my mistake!

**Comment by paulmarinas** [comment]
@erikn69 bro can you help me in setting up the team features I followed the docs and created middleware as well but setPermissionsTeamId is not working when creating user. I'm using filamentphp.

**Reply by erikn69** [reply]
> is not working when creating user

`setPermissionsTeamId` is not for user creation, it's for role/permission assignment. Also, I don't know filamentphp
